x/upgrade/keeper/keeper.go

```diff
@@ -243,24 +243,18 @@ func (k Keeper) GetLastCompletedUpgrade(ctx sdk.Context) (string, int64) {
 	iter := sdk.KVStoreReversePrefixIterator(ctx.KVStore(k.storeKey), []byte{types.DoneByte})
 	defer iter.Close()
 
-	var upgrades []upgrade
+	var latest upgrade
+	var found bool
 	for ; iter.Valid(); iter.Next() {
-		name := parseDoneKey(iter.Key())
 		value := int64(sdk.BigEndianToUint64(iter.Value()))
-
-		upgrades = append(upgrades, upgrade{Name: name, BlockHeight: value})
-	}
-
-	// sort upgrades in descending order by block height
-	sort.SliceStable(upgrades, func(i, j int) bool {
-		return upgrades[i].BlockHeight > upgrades[j].BlockHeight
-	})
-
-	if len(upgrades) > 0 {
-		return upgrades[0].Name, upgrades[0].BlockHeight
+		if !found || value >= latest.BlockHeight {
```

Comment:
Lets add a comment here that were searching for any upgrade that was past the current blockheight.

Reply:
I've renamed value -> upgradeHeight so I think now the code speaks for itself.